crates/ibc/Cargo.toml

```diff
@@ -46,6 +48,11 @@ derive_more = { version = "0.99.17", default-features = false, features = ["from
 uint = { version = "0.9", default-features = false }
 primitive-types = { version = "0.12.0", default-features = false, features = ["serde_no_std"] }
 dyn-clone = "1.0.8"
+displaydoc = {version = "0.2.3", default-features = false }
+## for codec encode or decode
+codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive"] }
+scale-info = { version = "2.1.2", default-features = false, features = ["derive"] }
+sp-runtime = {  version = "7.0.0", default-features = false }
```

Comment:
Can we make these [optional dependencies](https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies)? Something like:

```toml
[dependencies]
# ...
codec = {..., optional = true}
scale-info = {..., optional = true}
sp-runtime = {..., optional = true}

[features]
# ...
scale-codec = ["dep:codec", "dep:scale-info", "dep:sp-runtime"]
```